### ISSUE TITLE
Only half of the adversarial loss is optimised

### ISSUE THREAD

**Issue by MannyKayy** [edenton/drnet-py#3]
Hi,

Great Work !

It appears that only the first half of the adversarial loss (equation 3) is actually optimized with respect to the discriminator.

Although the sd_loss is calculated ([line 222](https://github.com/edenton/drnet-py/blob/master/train_drnet.py#L222)), the discriminator parameters are updated ([line 254](https://github.com/edenton/drnet-py/blob/master/train_drnet.py#L254)) after the gradients calculated by sd_loss are cleared ([line 235](https://github.com/edenton/drnet-py/blob/master/train_drnet.py#L235)). Thus it appears that the discriminator only updates with respect to the first [ce loss](https://github.com/edenton/drnet-py/blob/master/train_drnet.py#L251), i.e. equation 3.

**Comment by edenton** [comment]
You are correct the scene discriminator is only updated in the train_scene_discriminator function (not train(), which updates the rest of the model). Notice though, that the first have of the batch fed into the discriminator here: https://github.com/edenton/drnet-py/blob/master/train_drnet.py#L250 contains pose vectors from the same scene, and the second half contains pose vectors from different scenes. This corresponds to the two parts of the loss function.
